Week1/homework/app.js

```diff
@@ -2,10 +2,136 @@
+    'where_late_the_sweet_birds_sang',
+    'to_kill_a_mockingbird',
+    'the_help',
+    'never_let_me_go',
+    'gone_with_the_wind',
+    'it',
+    'interpreter_of_maladies',
+    '1Q84',
   ];
 
-  // Replace with your own code
-  console.log(bookTitles);
+  const mainHeading = document.createElement(`h1`);
+  mainHeading.setAttribute(`class`, `mainHeading`);
+  mainHeading.innerText = `My Favorite Books`;
+  document.body.appendChild(mainHeading);
+
+  function FavoriteBook(title, language, author) {
+    this.title = title;
+    this.language = language;
+    this.author = author;
+  }
+
+  // eslint-disable-next-line no-new-object
+  const favoriteBooks = new Object();
+
+  favoriteBooks[bookTitles[0]] = new FavoriteBook(`The Road`, `English`, `Cormac McCarthy`);
+  favoriteBooks[bookTitles[1]] = new FavoriteBook(
+    `One Hundred Years of Solitude`,
+    `English`,
+    `Gabriel Garcia Marquez`,
+  );
+  favoriteBooks[bookTitles[2]] = new FavoriteBook(
+    `Where Late the Sweet Birds Sang`,
+    `English`,
+    `Kate Wilhelm`,
+  );
+  favoriteBooks[bookTitles[3]] = new FavoriteBook(`To Kill a Mockingbird`, `English`, `Harper Lee`);
+  favoriteBooks[bookTitles[4]] = new FavoriteBook(`The Help`, `English`, `Kathryn Stockett`);
+  favoriteBooks[bookTitles[5]] = new FavoriteBook(`Never Let Me Go`, `English`, `Kazuo Ishiguro`);
+  favoriteBooks[bookTitles[6]] = new FavoriteBook(
+    `Gone With The Wind`,
+    `English`,
+    `Margaret Mitchell`,
+  );
+  favoriteBooks[bookTitles[7]] = new FavoriteBook(`It`, `English`, `Stephan King`);
+  favoriteBooks[bookTitles[8]] = new FavoriteBook(
+    `Interpreter of Maladies`,
+    `English`,
+    `Jhumpa Lahiri`,
+  );
+  favoriteBooks[bookTitles[9]] = new FavoriteBook(`1Q84`, `English`, `Haruki Murakami`);
+
+  function BookCover(coverImage) {
+    this.coverImage = coverImage;
+  }
+
+  // eslint-disable-next-line no-new-object
+  const favoriteBooksCovers = new Object();
+
+  favoriteBooksCovers[bookTitles[0]] = new BookCover(`images/the_road.jpg`);
+  favoriteBooksCovers[bookTitles[1]] = new BookCover(`images/one_hundred_years_of_solitude.jpg`);
+  favoriteBooksCovers[bookTitles[2]] = new BookCover(`images/where_late_the_sweet_birds_sang.jpg`);
+  favoriteBooksCovers[bookTitles[3]] = new BookCover(`images/to_kill_a_mockingbird.jpg`);
+  favoriteBooksCovers[bookTitles[4]] = new BookCover(`images/the_help.jpg`);
+  favoriteBooksCovers[bookTitles[5]] = new BookCover(`images/never_let_me_go.jpg`);
+  favoriteBooksCovers[bookTitles[6]] = new BookCover(`images/gone_with_the_wind.jpg`);
+  favoriteBooksCovers[bookTitles[7]] = new BookCover(`images/it.jpg`);
+  favoriteBooksCovers[bookTitles[8]] = new BookCover(`images/interpreter_of_maladies.jpg`);
+  favoriteBooksCovers[bookTitles[9]] = new BookCover(`images/1q84.jpg`);
+
+  let bookItem;
+  let bookDiv;
+
+  function createBookList() {
+    const bookList = document.createElement('ul');
+
+    for (const key of Object.keys(favoriteBooks)) {
```

Comment:
The idea of the instructions was to loop over the `bookTitles` array. I know that looping over the keys of `favoriteBooks` gives the same result, but be careful to follow the instructions.

Reply:
Ups you are right, I updated the code accordingly. Thanks for the correction.